### PR TITLE
Group tree module

### PR DESCRIPTION
**Pull request by jorgenherje**
Implementation of group tree module. Both front-end and back-end code.

Initial version has single realization functionality.

### CODE REVIEW

File: backend_py/primary/primary/services/group_tree_assembler/group_tree_assembler.py [hunk 1]
```diff
@@ -0,0 +1,943 @@
...
+        )
+        create_node_classifications_time_ms = timer.lap_ms()
+
+        # Initialize injection states based on group tree data
+        if self._terminal_node in node_classification_dict:
+            is_inj_in_grouptree = node_classification_dict[self._terminal_node].IS_INJ
+            if is_inj_in_grouptree and "FWIR" in vectors_table_column_names:
+                self._has_waterinj = pc.sum(single_realization_vectors_table["FWIR"]).as_py() > 0
+            if is_inj_in_grouptree and "FGIR" in vectors_table_column_names:
+                self._has_gasinj = pc.sum(single_realization_vectors_table["FGIR"]).as_py() > 0
```
[the remark this answers is not in the document]
Comment (lindjoha) [reply backend_py/primary/primary/services/group_tree_assembler/group_tree_assembler.py:229]: all_wi_vectors =  WWIR:{well}, GWIR:{group}
all_gi_vectors = WGIR:{well}, GGIR:{group}

**Review by lindjoha — COMMENTED** [review]
Very good! Ready to be merged after the changes we discussed on FWIR/FGIR.

File: backend_py/primary/primary/services/group_tree_assembler/_group_tree_dataframe_model.py [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+            # Filter out BRANPROP entries
+            self._grouptree_df = self._grouptree_df[self._grouptree_df["KEYWORD"] != TreeType.BRANPROP.value]
+        elif self._tree_type == TreeType.BRANPROP:
+            # Filter out GRUPTREE entries
+            self._grouptree_df = self._grouptree_df[self._grouptree_df["KEYWORD"] != TreeType.GRUPTREE.value]
+
+        group_tree_wells: set = set()
+        group_tree_groups: set = set()
+        group_tree_keywords = self._grouptree_df["KEYWORD"]
+        group_tree_nodes = self._grouptree_df["CHILD"]
```
[the remark this answers is not in the document]
Comment (jorgenherje) [reply backend_py/primary/primary/services/group_tree_assembler/_group_tree_dataframe_model.py:126]: They can be typed to list, and they are used to create list of group tree wells and group tree groups.

We cannot make the lists unique individually, as the columns `KEYWORD` and `CHILD` has a pair per index. I.e. keyword at an index determines whether the node is a well or group node. We can filter at dataframe beforehand, but then pandas basically performs the looping per row internally already. Thereby I suggest we just keep the logic as is.

File: backend_py/primary/primary/services/group_tree_assembler/group_tree_assembler.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorgenherje) [reply backend_py/primary/primary/services/group_tree_assembler/group_tree_assembler.py:246]: Check added. Looking for any WWIR or GWIR vectors among the vectors created for group tree and found in summary.
If any found, the FWIR is expected.

Same logic added for WGIR, GGIR and expected FGIR.

**Review by rubenthoms — APPROVED** [review]
Had a swift review and couldn't find anything that requires changes in this first version :+1:

**Review by lindjoha — APPROVED** [review]
I have looked mostly at the code in  `group_tree_assembler.py` and it is very good!

**Pull request merged**